src/tarantool-client/Converters/AuthenticationPacketConverter.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+
+using iproto.Data;
+using iproto.Data.Packets;
+
+using TarantoolDnx.MsgPack;
+
+namespace tarantool_client.Converters
+{
+    public class AuthenticationPacketConverter : IMsgPackConverter<AuthenticationPacket>
+    {
+        public void Write(AuthenticationPacket value, IBytesWriter writer, MsgPackContext context)
+        {
+            if (value == null)
+            {
+                context.NullConverter.Write(null, writer, context);
+            }
+            else
+            {
+                var headerConverter = context.GetConverter<Header>();
+                headerConverter.Write(value.Header, writer, context);
+
+                var bodyMapConverter = context.GetConverter<Dictionary<Key, object>>();
+                var bodyMap = new Dictionary<Key, object>()
```

Comment:
Also, methods for reading array and map headers extracted to IBytesReader.
HeaderConverter.Read and ResponsePacketConverter.Read methods rewriter using low-level methods.